library/std/src/sys/uefi/alloc.rs

```diff
@@ -0,0 +1,32 @@
+//! Global Allocator for UEFI.
+//! Uses [r-efi-alloc](https://crates.io/crates/r-efi-alloc)
+
+use crate::alloc::{handle_alloc_error, GlobalAlloc, Layout, System};
+
+const MEMORY_TYPE: u32 = r_efi::efi::LOADER_DATA;
+
+#[stable(feature = "alloc_system_type", since = "1.28.0")]
+unsafe impl GlobalAlloc for System {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let system_table = match crate::os::uefi::env::try_system_table() {
+            None => return crate::ptr::null_mut(),
+            Some(x) => x.as_ptr() as *mut _,
+        };
+
+        if layout.size() > 0 {
+            unsafe { r_efi_alloc::raw::alloc(system_table, layout, MEMORY_TYPE) }
+        } else {
+            layout.dangling().as_ptr()
+        }
+    }
+
+    unsafe fn dealloc(&self, ptr: *mut u8, layout: Layout) {
+        let system_table = match crate::os::uefi::env::try_system_table() {
+            None => handle_alloc_error(layout),
```

Comment:
`handle_alloc_error` will try to print a message, which will fail because there is no system table. It seems simpler to abort here, or to just assume this situation cannot occur, because there can be no allocations if this is reached, making the invocation of this function invalid to begin with.

Reply:
Well, this minimal std does not implement stdio anyway, so according to the `usupported/stdio.rs`, this should be fine since `None` will be returned for the panic_output.

Even my full std implementation returns `None` if system table is not found, so again, it shouldn't actually cause UB.